### PR TITLE
Generic for typed storage of configuration options and `Get_As_String` with default

### PR DESCRIPTION
**Pull request by mosteo**
These will serve to make a bit more robust the storage of config options in Alire.

### CODE REVIEW

File: src/clic-config.adb [hunk 1]
```diff
@@ -112,6 +112,23 @@ package body CLIC.Config is
...
+   -- Get_As_String --
+   -------------------
+
+   function Get_As_String (This    : Instance;
+                           Key     : Config_Key;
+                           Default : String) return String
+   is
+      Val : constant String := Get_As_String (This, Key);
+   begin
+      if Val = "" then
```
Comment (Fabien-Chouteau) [inline src/clic-config.adb:125]: I think this function should not use the other version of `Get_As_String` but the other way around
```ada
   -------------------
   -- Get_As_String --
   -------------------

   function Get_As_String (This    : Instance;
                           Key     : Config_Key;
                           Default : String) return String
   is
   begin
      if This.Defined (Key) then
         return Image (This.Get (Key).Value);
      else
         return Default;
      end if;
   end Get_As_String;

   -------------------
   -- Get_As_String --
   -------------------

    function Get_As_String (This : Instance;
                            Key : Config_Key)
                            return String
   is (Get_As_String (This, Key, Default => ""));
```

**Review by Fabien-Chouteau — CHANGES_REQUESTED** [review]

File: src/clic-config.adb [hunk 2]
```diff
@@ -100,18 +100,27 @@ package body CLIC.Config is
    -- Get_As_String --
    -------------------
 
-   function Get_As_String (This : Instance;
-                           Key : Config_Key)
-                           return String
+   function Get_As_String (This    : Instance;
+                           Key     : Config_Key;
+                           Default : String) return String
```
Comment (Fabien-Chouteau) [inline src/clic-config.adb:105]: Actually do we even need two function if the `Default` has a default value: `Default : String := ""` ? :)

**Review by Fabien-Chouteau — APPROVED** [review]

File: src/clic-config.adb (hunk 2 above)
Comment (mosteo) [reply src/clic-config.adb:105]: Haha right, indeed :)

**Comment by mosteo** [comment]
Third time is the charm?

**Pull request merged**